tests/basic/pool_function_test.go

```diff
@@ -859,3 +860,63 @@ var _ = Describe("{PoolExpandAddDiskInMaintenanceMode}", func() {
 		})
 	})
 })
+
+var _ = Describe("{StorageFullPoolExpansion}", func() {
+	var (
+		appList      []string
+		selectedNode *node.Node
+	)
+
+	BeforeEach(func() {
+		Inst().AppList = []string{"fio-fastpath-repl1"}
+		contexts = ScheduleApplications("storagefull-resize")
+		appList = Inst().AppList
+	})
+
+	JustBeforeEach(func() {
+		selectedNode = GetNodeWithLeastSize()
+		_ = Inst().S.AddLabelOnNode(*selectedNode, k8s.NodeType, k8s.FastpathNodeType)
+		log.FailOnError(err, fmt.Sprintf("Failed to add fastpath label on node %v", selectedNode.Name))
+	})
+
+	AfterEach(func() {
+		Inst().AppList = appList
+		appsValidateAndDestroy(contexts)
+		_ = Inst().S.RemoveLabelOnNode(*selectedNode, k8s.NodeType)
+	})
+
+	It("Expand pool with resize-disk type after pool is down due to storage full", func() {
+		// https://portworx.testrail.net/index.php?/cases/view/51280
+		StartTorpedoTest("StorageFullPoolExpandResizeType", "Expand pool with resize-disk type after pool is down due to storage full", nil, 51280)
+		Step("Prepare a full pool to expand", func() {
+			err = WaitForPoolOffline(*selectedNode)
+			log.FailOnError(err, fmt.Sprintf("Timed out waiting to load a pool and bring node %s storage down", selectedNode.Name))
+			poolsStatus, err := Inst().V.GetNodePoolsStatus(*selectedNode)
+			log.FailOnError(err, "error getting pool status on node %s", selectedNode.Name)
+			for i, s := range poolsStatus {
+				if s == "Offline" {
+					poolIDToResize = i
+					poolToResize, err = GetStoragePoolByUUID(poolIDToResize)
+					log.FailOnError(err, "error getting pool with UUID [%s]", poolIDToResize)
+					break
+				}
+			}
+		})
+
+		Step("Expand the full pool in type resize-disk", func() {
+			targetSizeGiB = (poolToResize.TotalSize / units.GiB) * 2
+			log.InfoD("Current Size of the pool %s is %d, trying to expand it to double the size", poolToResize.Uuid, poolToResize.TotalSize/units.GiB)
+			err = Inst().V.ExpandPool(poolToResize.Uuid, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, targetSizeGiB, true)
```

Comment:
We are using repl1 and skipWaitForCleanVolumes is true. This way the code that gets triggered is not an actual customer scenario. Is it? Do customers use forced option option when doing the  expand? For repl1 volumes if pool needs to be put to maintenance then they will need to stop IOs and detach the volumes, put pool explicitly in maintenance mode before doing the pool expand and this won't be an online operation right? 

Instead why don't we test with a repl2 app which triggers an actual prod scenario?

Reply:
This is a tradeoff we decided to take a few months ago. 
The functionality of `skipWaitForCleanVolumes` param is fairly simple and independent, we would test it working separately. In the meantime, for all other functional tests, we set it to true to avoid unnecessary complexity.